test/unit/org/apache/cassandra/tools/SSTableExportTest.java

```diff
@@ -43,12 +62,146 @@ public void testSSTableExport_NoArgs()
+            assertTrue(tool.getCleanedStderr(), tool.getCleanedStderr().isEmpty());
+            tool.assertOnExitCode();
+        }
+        assertPostTestEnv();
+    }
+
+    @Test
+    public void testPKArg() throws IOException
+    {
+        try (ToolRunner tool = runner.invokeClassAsTool("org.apache.cassandra.tools.SSTableExport", findOneSSTable("legacy_sstables", "legacy_ma_simple"), "-k", "0"))
+        {
+            List<Map<String, Object>> parsed = mapper.readValue(tool.getStdout(), jacksonListOfMapsType);
+            assertEquals(tool.getStdout(), 1, parsed.size());
+            assertEquals(tool.getStdout(), "0", ((List) ((Map) parsed.get(0).get("partition")).get("key")).get(0));
+            assertTrue(tool.getCleanedStderr(), tool.getCleanedStderr().isEmpty());
+            tool.assertOnExitCode();
+        }
+        assertPostTestEnv();
+    }
+
+    @Test
+    public void testExcludePKArg() throws IOException
+    {
+        try (ToolRunner tool = runner.invokeClassAsTool("org.apache.cassandra.tools.SSTableExport", findOneSSTable("legacy_sstables", "legacy_ma_simple"), "-x", "0"))
+        {
+            List<Map<String, Object>> parsed = mapper.readValue(tool.getStdout(), jacksonListOfMapsType);
+            assertEquals(tool.getStdout(), 4, parsed.size());
+            assertTrue(tool.getCleanedStderr(), tool.getCleanedStderr().isEmpty());
+            tool.assertOnExitCode();
+        }
+        assertPostTestEnv();
+    }
+
+    @Test
+    public void testTSFormatArg() throws IOException
+    {
+        try (ToolRunner tool = runner.invokeClassAsTool("org.apache.cassandra.tools.SSTableExport", findOneSSTable("legacy_sstables", "legacy_ma_simple"), "-t"))
+        {
+            assertThat(tool.getStdout(),
+                       CoreMatchers.both(CoreMatchers.containsString("1445008632854000"))
+                                   .and(CoreMatchers.not(CoreMatchers.containsString("2015-10-16T15:17:13.030Z"))));
+            assertTrue(tool.getCleanedStderr(), tool.getCleanedStderr().isEmpty());
+            tool.assertOnExitCode();
+        }
+        assertPostTestEnv();
+    }
+
+    @Test
+    public void testJSONLineArg() throws IOException
+    {
+        try (ToolRunner tool = runner.invokeClassAsTool("org.apache.cassandra.tools.SSTableExport", findOneSSTable("legacy_sstables", "legacy_ma_simple"), "-l"))
+        {
+                try
+                {
+                    mapper.readValue(tool.getStdout(), jacksonListOfMapsType);
+                    fail("Shouldn't be able to deserialize that output, now it's not a collection anymore.");
+                }
+                catch(MismatchedInputException e)
+                {
+                }
+
+                int parsedCount = 0;
+                for (String line : tool.getStdout().split("\\R"))
+                {
+                    mapper.readValue(line, Map.class);
```

Comment:
this only validates that this is a object, should also validate the content.

Reply:
Tricky as each line is different but I added a check for one of the keys.